exercises/hamming/test/hamming_test.dart

```diff
@@ -1,84 +1,58 @@
-import 'package:test/test.dart';
 import 'package:hamming/hamming.dart';
+import 'package:test/test.dart';
+
+final hamming = Hamming();
 
 /// Test cases match one-to-one with those described
 /// in the problem specification found here:
 /// https://github.com/exercism/problem-specifications/blob/master/exercises/hamming/canonical-data.json
 void main() {
-  final Hamming hamming = new Hamming();
+  final mustEqual = predicate(
+      (ArgumentError e) => e is ArgumentError && e.message == 'left and right strands must be of equal length',
```

Comment:
I do not understand this. If the argument to predicate already has type `ArgumentError`. There is no need for `e is ArgumentError`. Since `predicate` has generic parameters, type annotation on the function parameter is required, otherwise it is inferred as `dynamic`. Since the `predicate` is being used as a generic matcher inside a `throwsA` matcher, the type annotation for function argument e should be removed. And `e is ArgumentError` should be kept.

```suggestion
      (e) => e is ArgumentError && e.message == 'left and right strands must be of equal length',
```